### ISSUE TITLE
Placeholders

### ISSUE THREAD

**Issue by ProJakob** [andrei1058/BedWarsProxy#65]
When someone joins my server a scoreboard with their stats is displayed using DeluxeHub. The problem is that the server hangs for 1-2 seconds when requesting the placeholder.

**Comment by andrei1058** [comment]
you should post this on the bedwars proxy repository https://github.com/andrei1058/BedWarsProxy

**Comment by andrei1058** [comment]
Could you please list the placeholders in question?

**Comment by ProJakob** [comment]
Sorry for the late response, I was having issues when using these placeholders in the scoreboard:
`%bw1058_player_level%`
`%bw1058_player_xp_formatted%`
`%bw1058_player_rerq_xp_formatted%`
`%bw1058_player_progress%`
`%bw1058_stats_gamesplayed%`
`%bw1058_stats_total_kills%`
`%bw1058_stats_wins%`

Could the lag be caused because the stats are loaded from the database when the placeholder is requested?
https://github.com/andrei1058/BedWarsProxy/blob/development/proxy-plugin/src/main/java/com/andrei1058/bedwars/proxy/database/StatsCache.java#L276

**Comment by andrei1058** [comment]
Yes, lag can be caused by mysql queries if triggered every time the placeholder is requested.
I'm not sure why the class is called StatsCache with all those queries in it. Will have to look into it with the next milestone.

**Comment by ProJakob** [comment]
Alright Thanks! My Solution for now is to just cache the placeholder value myself which seems to work fine.
